threedi_modelchecker/config.py

```diff
@@ -1561,6 +1569,129 @@ def is_none_or_empty(col):
+        ),
+        message="v2_vegetation_drag.height must be defined.",
+    ),
+    QueryCheck(
+        error_code=503,
+        level=CheckLevel.WARNING,
+        column=models.VegetationDrag.height,
+        invalid=Query(models.VegetationDrag).filter(
+            vegetation_drag_filter,
+            models.VegetationDrag.height == None,
+            ~is_none_or_empty(models.VegetationDrag.height_file),
+        ),
+        message="v2_vegetation_drag.height is recommended as fallback value when using a height_file.",
+    ),
+    RangeCheck(
+        error_code=504,
+        column=models.VegetationDrag.stem_count,
+        filters=vegetation_drag_filter,
+        min_value=0,
+        left_inclusive=False,
+    ),
+    QueryCheck(
+        error_code=505,
+        column=models.VegetationDrag.stem_count,
+        invalid=Query(models.VegetationDrag).filter(
+            vegetation_drag_filter,
+            models.VegetationDrag.stem_count == None,
+            is_none_or_empty(models.VegetationDrag.stem_count_file),
+        ),
+        message="v2_vegetation_drag.stem_count must be defined.",
+    ),
+    QueryCheck(
+        error_code=506,
+        level=CheckLevel.WARNING,
+        column=models.VegetationDrag.stem_count,
+        invalid=Query(models.VegetationDrag).filter(
+            vegetation_drag_filter,
+            models.VegetationDrag.stem_count == None,
+            ~is_none_or_empty(models.VegetationDrag.stem_count_file),
+        ),
+        message="v2_vegetation_drag.stem_count is recommended as fallback value when using a stem_count_file.",
+    ),
+    RangeCheck(
+        error_code=507,
+        column=models.VegetationDrag.stem_diameter,
+        filters=vegetation_drag_filter,
+        min_value=0,
+        left_inclusive=False,
+    ),
+    QueryCheck(
+        error_code=508,
+        column=models.VegetationDrag.stem_diameter,
+        invalid=Query(models.VegetationDrag).filter(
+            vegetation_drag_filter,
+            models.VegetationDrag.stem_diameter == None,
+            is_none_or_empty(models.VegetationDrag.stem_diameter_file),
+        ),
+        message="v2_vegetation_drag.stem_diameter must be defined.",
+    ),
+    QueryCheck(
+        error_code=509,
+        level=CheckLevel.WARNING,
+        column=models.VegetationDrag.stem_diameter,
+        invalid=Query(models.VegetationDrag).filter(
+            vegetation_drag_filter,
+            models.VegetationDrag.stem_diameter == None,
+            ~is_none_or_empty(models.VegetationDrag.stem_diameter_file),
+        ),
+        message="v2_vegetation_drag.stem_diameter is recommended as fallback value when using a stem_diameter_file.",
+    ),
+    RangeCheck(
+        error_code=510,
+        column=models.VegetationDrag.drag_coefficient,
+        filters=vegetation_drag_filter,
+        min_value=0,
+        left_inclusive=False,
+    ),
+    RangeCheck(
+        error_code=510,
+        level=CheckLevel.WARNING,
```

Comment:
So to be sure:

- an error if drag_coefficient <= 0
- a warning if drag_coefficient <= 1